lib/bolt/node/ssh.rb

```diff
@@ -14,6 +14,16 @@ def self.initialize_transport(logger)
       }
     end
 
+    if Gem.win_platform?
+      require 'ffi'
+      module Win
+        extend FFI::Library
+        ffi_lib 'user32'
+        ffi_convention :stdcall
+        attach_function :FindWindow, :FindWindowA, %i[string string], :int
```

Comment:
Should use `FindWindowW`, not `FindWindowA` ... which means using wide strings.

Reply:
I initially tried FindWindowW:

```
attach_function :FindWindowW, :FindWindowW, %i[buffer_in buffer_in], :int
```
But it always returns 0, even when `FindWindowA` succeeds.

Any advice?

Reply:
You basically need to do an `.encode('UTF-16LE')` on the string to make sure its a wide string before passing. We implement some helpers in Puppet itself, but probably can't rely on that here - https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/api_types.rb#L23-L34

Still, it seems getting a process list, rather than searching by window name is preferred? I'm not sure why `pageant` has a window?

Reply:
Using `encode` works, thank you! I'll push tomorrow and then look at searching the process list, but have been avoiding deviating from net-ssh's tests.

Reply:
FWIW, this is how putty itself checks for the `pageant`'s presence: https://github.com/github/putty/blob/49fb598b0e78d09d6a2a42679ee0649df482090e/windows/winpgntc.c#L18-L26
So using the `FindWindowW` seems appropriate.